web/src/routes/CategoriesPage.tsx

```diff
@@ -23,19 +28,47 @@ import CityContentLayout, { CityContentLayoutProps } from '../components/CityCon
 import CityContentToolbar from '../components/CityContentToolbar'
 import FailureSwitcher from '../components/FailureSwitcher'
 import Helmet from '../components/Helmet'
-import LoadingSpinner from '../components/LoadingSpinner'
+import PageSkeleton from '../components/PageSkeleton'
+import SkeletonHeader from '../components/SkeletonHeader'
+import SkeletonList from '../components/SkeletonList'
 import buildConfig from '../constants/buildConfig'
 import { cmsApiBaseUrl } from '../constants/urls'
 import usePreviousProp from '../hooks/usePreviousProp'
 import useTtsPlayer from '../hooks/useTtsPlayer'
 
 const CATEGORY_NOT_FOUND_STATUS_CODE = 400
+const NUM_CARD_SKELETONS = 16
 
 const getBreadcrumb = (category: CategoryModel, cityName: string): BreadcrumbProps => ({
   title: category.isRoot() ? cityName : category.title,
   to: category.path,
 })
 
+const StyledBox = styled(Box)`
+  display: grid;
+  gap: 32px 24px;
+  grid-template-columns: repeat(auto-fit, minmax(160px, 1fr));
+  justify-content: center;
+  width: 100%;
+`
+
+const StyledCard = styled(Card)`
+  width: 160px;
+`
+
+const RootCategorySkeleton = (): ReactElement => (
+  <Stack paddingTop={2} alignItems='center'>
+    <SkeletonHeader />
+    <StyledBox>
+      {[...Array(NUM_CARD_SKELETONS).keys()].map(index => (
```

Comment:
here's another way without spreading:
```suggestion
      {Array.from({ length: NUM_CARD_SKELETONS }, (_, index) => (
```